mealie/services/scheduler/tasks/post_webhooks.py

```diff
@@ -1,54 +1,65 @@
 from datetime import datetime, timezone
+from typing import Optional
 
-import requests
-from fastapi.encoders import jsonable_encoder
 from pydantic import UUID4
-from sqlalchemy.orm import Session
 
 from mealie.db.db_setup import create_session
-from mealie.db.models.group.webhooks import GroupWebhooksModel
 from mealie.repos.all_repositories import get_repositories
+from mealie.schema.response.pagination import PaginationQuery
+from mealie.services.event_bus_service.event_bus_service import EventBusService
+from mealie.services.event_bus_service.event_types import (
+    INTERNAL_INTEGRATION_ID,
+    EventDocumentType,
+    EventOperation,
+    EventTypes,
+    EventWebhookData,
+)
 
 last_ran = datetime.now(timezone.utc)
 
 
-def get_scheduled_webhooks(session: Session, bottom: datetime, top: datetime) -> list[GroupWebhooksModel]:
-    """
-    get_scheduled_webhooks queries the database for all webhooks scheduled between the bottom and
-    top time ranges. It returns a list of GroupWebhooksModel objects.
-    """
-
-    return (
-        session.query(GroupWebhooksModel)
-        .where(
-            GroupWebhooksModel.enabled == True,  # noqa: E712 - required for SQLAlchemy comparison
-            GroupWebhooksModel.scheduled_time > bottom.astimezone(timezone.utc).time(),
-            GroupWebhooksModel.scheduled_time <= top.astimezone(timezone.utc).time(),
-        )
-        .all()
-    )
+def post_group_webhooks(start_dt: Optional[datetime] = None, group_id: Optional[UUID4] = None) -> None:
+    """Post webhook events to specified group, or all groups"""
 
-
-def post_group_webhooks() -> None:
     global last_ran
-    session = create_session()
-    results = get_scheduled_webhooks(session, last_ran, datetime.now())
 
-    last_ran = datetime.now(timezone.utc)
+    end_dt = datetime.now(timezone.utc)
+    if start_dt is None:
+        # set the webhook query bounds to start at the last time the service ran
+        start_dt = last_ran
 
-    repos = get_repositories(session)
+    # update the last ran time so we continue here next time the service runs
+    last_ran = end_dt
 
```

Comment:
I think this could be simplified into something like this...

```python
last_ran = end_dt = datetime.now(timezone.utc)
stard_dt = start_dt or last_ran
```

Mostly personal preference, ¯\_(ツ)_/¯

Reply:
Ah yeah I like this much better. This also lets me set start _then_ end, which is also nice